x-pack/plugins/infra/public/pages/metrics/hosts/utils.ts

```diff
@@ -5,16 +5,23 @@
  * 2.0.
  */
 
-import { Filter } from '@kbn/es-query';
-import { SnapshotNode } from '../../../../common/http_api';
+import { DataViewBase, Filter } from '@kbn/es-query';
 
-export const createHostsFilter = (hostNodes: SnapshotNode[]): Filter => {
+export const createHostsFilter = (hostNames: string[], dataView?: DataViewBase): Filter => {
   return {
     query: {
       terms: {
-        'host.name': hostNodes.map((p) => p.name),
```

Comment:
**doubt**: do we really need to move this mapping out of here and repeat it on each `createHostsFilter` call?
What is the purpose of lifting this mapping out of the function implementation?

Reply:
I had to do this because here[https://github.com/elastic/kibana/pull/155249/files#diff-2f6eebf2a5a7374b64991ceb0cd55fd2fe8e50361b8d0092af2b972027495fb8R65](url), the `currentPage` type is `HostNodeRow` and not `SnapshotNode`.